artemis-jakarta-client/pom.xml

```diff
@@ -0,0 +1,64 @@
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>org.apache.activemq</groupId>
+        <artifactId>artemis-pom</artifactId>
+        <version>2.17.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>artemis-jakarta-client</artifactId>
+    <packaging>jar</packaging>
+    <name>ActiveMQ Artemis JakartaEE Messaging Client</name>
+
+    <properties>
+        <activemq.basedir>${project.basedir}/..</activemq.basedir>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-jms-client</artifactId>
+            <version>${project.version}</version>
+            <scope>provided</scope>
+        </dependency>
+    </dependencies>
```

Comment:
Feels a bit odd that this module doesn't seem to actually end up with a dependency on the new API, shouldnt it? Actually, what about all the other client deps too for that matter, aren't they are omitted also...how would someone use this module currently?

Reply:
The produced jar should 'replace' the artemis-jms-client jar in the classpath. This dependency is there to 'force' the reactor to build the jms client first.

Reply:
I got that the dep is needed for reactor ordering, but I dont see it makes sense that the produced artemis-jakarta-client would seemingly have no dependencies specified. If so how is someone meant to actually use it, i.e how do they know what the rest of the classpath required for the client is actually meant to be in order to successfully use it? Depend on both client modules and overlook the extra dep(s) (I guess at least the API could be manually excluded)?

Reply:
Well ten a bom would make sense somehow. The goal was to provide a drop in replacement jar. Somehow, all the dependencies of the JMS client are required except for the API change. I'm not sure the goal of a pom is to provide a runtime classpath ... Maybe we could add some doc on how to use it ? Or a bom or a jakarta all client ?

Reply:
I'd say it is often expected a pom will provide the needed classpath, and would likely be in this case given historic usage. That is to say I expect few people declaring a dependency on artemis-jms-client are surprised it yields one, and guess a good proportion who might later try artemis-jakarta-client would be surprised it didn't similarly. I'm not sure how a bom would help here, unless the thought is someone could manually copy the needed dep definitions from its contents? An 'all' jar could certainly be an option (though I don't like those personally).

Perhaps its fine as a first step, but yes some doc (e.g at least a README in the module) on how to use it would seem needed, I think people would struggle to use it as-is otherwise.

Reply:
Added the dependencies in the runtime scope